pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,223 @@ func TestAccessTokenFile(t *testing.T) {
+		Convey("Given access token with data and a file location defined", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite2"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
-
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Second time
-			dataActual = []byte{'t', 'e', 's', 't', '2'}
-			err = tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Test we are reading the new Data
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
-
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Writes the access token without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+			})
+
+			Convey("When running the Write method a second time", func() {
+				dataActual = []byte{'t', 'e', 's', 't', '2'}
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Writes the data the file as expected", func() {
+					dataExpected, _ := tokenInFile.Read()
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 	})
 
 	Convey("Delete", t, func() {
-		Convey("Returns no error after read write and delete", func() {
-			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
-
-			dataFromRead, err := tokenInFile.Read()
-			So(err, ShouldEqual, nil)
-
-			err = tokenInFile.Delete()
-			So(err, ShouldEqual, nil)
 
-			// Check that file does not exits
-			_, err = os.Stat("/tmp/accessTokenFileDel1")
-			So(err, ShouldNotEqual, nil)
+		Convey("Given access token with data", func() {
+			dataActual := []byte{'t', 'e', 's', 't'}
 
-			// Both input & output arrays should be cleared from memory
-			empty := make([]byte, len(dataActual))
-			eq := reflect.DeepEqual(dataActual, empty)
-			So(eq, ShouldEqual, true)
-			eq = reflect.DeepEqual(dataFromRead, empty)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
```

Comment:
this can go inside the convey above.
```
Convey("And the data was written successfully", func() {
	tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel1"
	err := tokenInFile.Write(dataActual)
	So(err, ShouldEqual, nil)
})
```

Reply:
so not `Convey("When running the Write method", func() {...`?

Reply:
yes